packages/x-date-pickers/src/internals/hooks/useField/useField.interfaces.ts

```diff
@@ -173,13 +174,22 @@ interface FieldActiveDateManager<TValue, TDate> {
   ) => Pick<UseFieldState<TValue, any>, 'value' | 'referenceValue'>;
 }
 
-export type FieldSelectedSectionsIndexes = { startIndex: number; endIndex: number };
+export type FieldSelectedSectionsIndexes = {
+  startIndex: number;
+  endIndex: number;
+  /**
+   * If `true`, the selectors at the very beginning and very end of the input will be selected.
+   * @default false
+   */
+  shouldSelectBoundarySelectors?: boolean;
+};
 
 export type FieldSelectedSections =
   | number
   | FieldSelectedSectionsIndexes
   | MuiDateSectionName
-  | null;
+  | null
+  | 'all';
```

Comment:
I try to select the separators that are at the very beginning and end of the input only when we do Ctrl + A, or focus.

My initial check was `selectedSectionIndexes.startIndex === 0 && selectedSectionIndexes.endIndex === state.sections.length - 1`
But it does not work on single section format.

If you have a better idea, I'm interested.
It feels weird if we don't select them when doing Ctrl + A
It feels weird if we select them when clicking on the section in a single section format

Reply:
https://user-images.githubusercontent.com/3309670/207849408-474f1545-7378-4d2c-980d-005bfe501d3f.mp4

Reply:
I don't get what's going wrong in your recording. For me the following are correct behavior

- click on before/after put the select on the associated placeholder
- ctrl+a select the entire value

Reply:
Introducing `all` does not look like a problem


Instead of `shouldSelectBoundarySelectors: true` you could have `includesStartSparetor` and `includesEndSparetor` for more flexibility.

The only limitation I see is selecting only the first date, because I guess the current strategy allows to select:
- `YYYY`
- `Before YYYY After - ` 
- but not `Before YYYY After`

Reply:
I thought about the empty section, the keyboard can be managed by modifying the `getSectionOrder`, it would also require to modify the function that select a section on click to ignore it.

Another solution could be to remove it from sections, and only add its content when generating the input value from sections, and computing sections positions.

But both sound too hacky

Reply:
> Another solution could be to remove it from sections, and only add its content when generating the input value from sections, and computing sections positions.

Could be the best solution
We can experiment later on